File: 0383-ransom-note/0383-ransom-note.cpp
#include <unordered_map>
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> dict;

        for (char val : magazine) {
            dict[val]++;
        }

        for (char val : ransomNote) {
            if (dict.find(val) != dict.end()) {
                if (!dict[val]) {
                    return false;
                }
                dict[val]--;
            } else {
                return false;
            }
        }

        return true;
    }
};

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        letters: List[int] = [0]*26
        val: int = 0
        for letter in magazine:
            val = ord(letter)-97
            letters[val]+=1
        for letter in ransomNote:
            val = ord(letter)-97
            if letters[val] > 0:
                letters[val]-=1
                if letters[val] < 0:
                    return False
            else:
                return False
        return True